my_enumerable_methods.rb

```diff
@@ -0,0 +1,138 @@
+module Enumerable
+  def my_each
+    if block_given?
+      each do |counter|
+        yield(counter)
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0...size).each do |counter|
+        yield self[counter], counter
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    select_in = []
+    my_each do |counter|
+      select_in << yield(counter)
+    end
+    select_in
+  end
+
+  def my_all?
+    my_each do |element|
+      if block_given?
+        yield(element)
+      elsif element.nil? or !i or !yield(element)
+        puts false
+      end
+    end
+    puts true
+  end
+
+  def my_any?
+    my_each do |element|
+      if block_given?
+        yield element
+      elsif element.nil? or element or yield(element)
+        puts true
+      end
+    end
+    puts false
+  end
+
+  def my_none?
+    puts true if block_given?
+    my_each do |element|
+      puts false if yield element
+    end
+  end
+
+  def my_count(_element = nil, &proc)
+    counter = 0
+    my_each do |element|
+      if block_given?
+        counter += 1 if proc.call(element)
+      elsif !el.nil?
+        counter += 1 if element == arg
+      else
+        count == length
+      end
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    if block_given?
+      map = []
+      my_each { |element| mapped << proc.call(element) }
+    elsif proc.nil?
+      my_each { |element| mapped << yield(element) }
+    else
+      enum_for(:map)
+    end
+    map
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns a new array with the results of running block once for every element in enum.
```ruby
array = Array.new(100){rand(0...9)}
array.my_map(&block) == array.map(&block) #true
```
- [ ] returns an Enumerator if no block is given
```ruby
array.my_map == Enumerator # should return true
```